src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -0,0 +1,802 @@
+                                    dependencyName = new StringBuilder();
+                                    dependencyChar++;
+                                    if (dependencyNameStr.StartsWith("+", StringComparison.Ordinal))
+                                    {
+                                        // this entry is actually a service load group
+                                        Interop.ENUM_SERVICE_STATUS_PROCESS[] loadGroup = GetServicesInGroup(_machineName, dependencyNameStr.Substring(1));
+                                        foreach (Interop.ENUM_SERVICE_STATUS_PROCESS groupMember in loadGroup)
+                                        {
+                                            if (!dependencyHash.ContainsKey(groupMember.serviceName))
+                                                dependencyHash.Add(groupMember.serviceName, new ServiceController(MachineName, groupMember));
+                                        }
+                                    }
+                                    else
+                                    {
+                                        if (!dependencyHash.ContainsKey(dependencyNameStr))
+                                            dependencyHash.Add(dependencyNameStr, new ServiceController(dependencyNameStr, MachineName));
+                                    }
+                                }
+                            }
+                        }
+
+                        _servicesDependedOn = new ServiceController[dependencyHash.Count];
+                        dependencyHash.Values.CopyTo(_servicesDependedOn, 0);
+
+                        return _servicesDependedOn;
+                    }
+                    finally
+                    {
+                        Marshal.FreeHGlobal(bufPtr);
+                    }
+                }
+                finally
+                {
+                    Interop.mincore.CloseServiceHandle(serviceHandle);
+                }
+            }
+        }
+
+        public SafeHandle ServiceHandle
+        {
+            get
+            {
+                return new SafeServiceHandle(GetServiceHandle(Interop.SERVICE_ALL_ACCESS));
+            }
+        }
+
+        /// Gets the status of the service referenced by this object, e.g., Running, Stopped, etc.
+        public ServiceControllerStatus Status
+        {
+            get
+            {
+                GenerateStatus();
+                return _status;
+            }
+        }
+
+        /// Gets the type of service that this object references.
+        public ServiceType ServiceType
+        {
+            get
+            {
+                GenerateStatus();
+                return (ServiceType)_type;
+            }
+        }
+
+        private static bool CheckMachineName(string value)
+        {
+            if (value == null)
+                return false;
+
+            value = value.Trim();
+            if (value.Equals(String.Empty))
+                return false;
```

Comment:
Seems like this whole method could be turned into the following (and avoid the potential allocations/copies in Trim):
```C#
private static bool CheckMachineName(string value)
{
    return !string.IsNullOrWhitespace(value) && value.IndexOf('\\') == -1;
}
```